### PR TITLE
Use proxy_ssl_context as SSL ctx when forwarding (PR V2)

### PR DESCRIPTION
**Pull request by abebeos**
rebased and slightly updated PR

- #2651 (by @ avi364)

~Some tests failing.~

### CODE REVIEW

File: src/urllib3/connection.py [hunk 1]
```diff
@@ -650,6 +665,11 @@ def connect(self) -> None:
         # Remove trailing '.' from fqdn hostnames to allow certificate validation
         server_hostname_rm_dot = server_hostname.rstrip(".")
 
+        if self._has_connected_to_proxy and self.proxy_config:
```
Comment (abebeos) [inline src/urllib3/connection.py:668]: see https://github.com/urllib3/urllib3/pull/2651/files#r938100111

Comment (abebeos) [reply src/urllib3/connection.py]: => changed this to use `proxy_is_forwarding` property

File: src/urllib3/connection.py [hunk 2]
```diff
@@ -551,7 +551,22 @@ def __init__(
...
+                )
+            elif proxy_config.ssl_context is None and ssl_context is not None:
+                raise ValueError(
+                    "The 'ssl_context' parameter cannot be set when use_forwarding_for_https is True. Use the 'proxy_ssl_context' parameter to customize the connection to the proxy"
+                )
+            else:
+
+                self.ssl_context = proxy_config.ssl_context
+        else:
+            self.ssl_context = ssl_context
```
Comment (abebeos) [inline src/urllib3/connection.py:568]: see https://github.com/urllib3/urllib3/pull/2651/files#r938100724

Comment (abebeos) [reply src/urllib3/connection.py:566]: => restore original logic within [c49e331](https://github.com/urllib3/urllib3/pull/3295/commits/c49e331ecc493a8cd16d54a086965d0c0aea3e2c)

**Comment by sigmavirus24** [comment]
@abebeos GitHub will automatically link your pull request when you reference it, you do not need to create additional comments [like this one](https://github.com/urllib3/urllib3/pull/2651#issuecomment-1902607601) as they only serve to generate duplicate notifications that create noise for those looking for signal in watching urllib3. I understand you have a very different way of working, but you're working on urllib3 and if you wish to continue contributing, we need you to be more respectful of the other maintainers and contributors involved (including not opening issues for pull requests you haven't written tests for and have not been merged; and commenting superfluously on pull requests you've chosen to try to rebase and get into a mergeable state). If you cannot adjust appropriately, consider moving on from urllib3 to projects that better align with your style and priorities.

**Comment by abebeos** [comment]
> (including not opening issues for pull requests you haven't written tests for and have not been merged; and commenting superfluously on pull requests you've chosen to try to rebase and get into a mergeable state)

Wow, you're super-confusing things. Just reread after 48hours.

@sethmlarson , this PR should be ready.

**Comment by sigmavirus24** [comment]
@abebeos the tests are failing. There's no point to review something before the tests are passing. Please ensure the tests pass before asking for reviews

**Comment by abebeos** [comment]
> the tests are failing. There's no point to review something before the tests are passing. Please ensure the tests pass before asking for reviews

This is complete nonsese.

@sethmlarson , why is this 'virus' allowed to harass other contributors on this repo? Or is it a chatbot to keep lonely devs company when traffic is low?

**Comment by sigmavirus24** [comment]
Failing PyPy tests on related testcases

@abebeos here is an example of how the tests are presently failing on a related test case. Again, if you need someone to coach you on how to go through and find what's wrong with your PR in the CI logs, we can do that, but it seems like you don't welcome feedback or requests for you to conform to the processes here.

As it stands, I'm pointing out how your behavior is at odds with this project and providing you feedback on how to better work with us. If you consider that harassment, that's disappointing but also a good sign that this is not a fit for you.

As it stands, I'm the only one continuing to try to help you work with us in a positive fashion. If you want to continue to demean and mock other contributors here, I believe the best course of action will be to prevent further contributions and interactions from you. I doubt the other maintainers will disagree.

**Comment by sethmlarson** [comment]
[These comments](https://github.com/urllib3/urllib3/pull/3295#issuecomment-1902644613) aren't acceptable and you've shown repeatedly that you're unwilling to change your behavior. For now you'll be banned from contributing to this repository.

**Pull request closed without merging**